src/murmur/Meta.cpp

```diff
@@ -125,6 +125,25 @@ T MetaParams::typeCheckedFromSettings(const QString &name, const T &defaultValue
 	return cfgVariable.value<T>();
 }
 
+bool MetaParams::hasDualStackSupport() {
+#ifdef Q_OS_UNIX
+	int s = ::socket(AF_INET6, SOCK_STREAM, 0);
+	if (s != -1) {
+		int ipv6only = 0;
+		socklen_t optlen = sizeof(ipv6only);
+		if (getsockopt(s, IPPROTO_IPV6, IPV6_V6ONLY, &ipv6only, &optlen) == 0) {
+			if (ipv6only == 0)
+				return true;
+		}
+		close(s);
+	}
+	return false;
+#else
+	// Assume dual-stacked for non-unix systems
+	return true;
```

Comment:
This assumption may not hold on all our supported platforms. See http://stackoverflow.com/questions/2108746/dual-socket-for-windows . How do we best handle this?